week1/index.js

```diff
@@ -1,25 +1,62 @@
-// To run this file: node index.js
-// Then in your browser: http://localhost:8080
+// Determine the port and the initial value of state
+var state = 10;
+var port = 8080;
+
+// Start the HTTP server
 var http = require('http');
+var server = http.createServer(handleRequest);
 
-var port = 8080;
 
-var server = http.createServer();
+// This function is the handler for "response"
+// It sends the requested information to the user
+function handleResponse(response, statusCode, h1Text, pText) {
+    response.writeHead(statusCode, {'content-type': 'text/html'});
+    response.write('<html><head></head><body><h1>' + h1Text + '</h1><p>' + pText + '</p></body></html>');
+    response.end();// End the response.
+}
 
-// Start the HTTP server, start listening for requests
-server.listen(port, function(error) {
-  if (error) {
-    console.log(error);
-  } else {
-    console.log('api listening on port', port);
-  }
-});
+// This function is the handler for "request"
+// It understands the request, processes it
+// and then sends it to the handler for "response"
+function handleRequest(request, response) {
+    console.log('New http request received', request.url);
+    var statusCode = 200;// Set the default status code.
```

Comment:
Cool use of defaults, very sensible. You could also use `let` here, to make sure they don't accidentally get written in a different scope.